python/golem_task_api/client.py

```diff
@@ -134,43 +134,38 @@ def __init__(
 class ProviderAppClient:
     DEFAULT_PORT: ClassVar[int] = 50006
 
-    @classmethod
-    def _start_service(
-            cls,
+    def __init__(
+            self,
             service: TaskApiService,
-    ) -> ProviderAppStub:
-        host, port = service.start(
-            f'provider {cls.DEFAULT_PORT}',
-            cls.DEFAULT_PORT,
-        )
-        return ProviderAppStub(
+            port: int = DEFAULT_PORT,
+    ) -> None:
+        self._service = service
+        host, port = service.start(f'requestor {port}', port)
+        self._golem_app = ProviderAppStub(
             Channel(host, port, loop=asyncio.get_event_loop()),
         )
 
-    @classmethod
     async def compute(
-            cls,
-            service: TaskApiService,
+            self,
             task_id: str,
             subtask_id: str,
             subtask_params: dict,
     ) -> Path:
-        golem_app = cls._start_service(service)
         request = ComputeRequest()
         request.task_id = task_id
         request.subtask_id = subtask_id
         request.subtask_params_json = json.dumps(subtask_params)
-        reply = await golem_app.Compute(request)
-        await service.wait_until_shutdown_complete()
+        reply = await self._golem_app.Compute(request)
```

Comment:
What will happen on this line if `shutdown` is called during computation? I think we should probably return `CANCELLED` or `ABORTED` status.

Reply:
I second that question. But simply throwing is fine too imo.

Reply:
i'm not sure what exactly happens here, but i think its forgotten about and that seems OK.
This function returns a result path or raise an exception

In golem core [this catch](https://github.com/golemfactory/golem/pull/4564/files#diff-2d82676f75ac94238cc5657493873ce6R323) will trigger when shutdown is called, if we set any status i think it should be there

Reply:
Please find out what happens exactly - you can use blenderapp for testing.

There's non zero chance that the server may wait for all the pending requests to complete before shutting down, so let's not be taking a guess here.

Reply:
I think we should not rely on the fact the result is forgotten about in Golem code. Task API is a separate library and even though it is highly unlikely that this part of the code will be used by anyone outside of Golem we should still act as we're caller-agnostic.

That being said I think it should be established what kind of error is being thrown in the described situation, catch this error and return a `CANCELLED` status.